### PR TITLE
fix: prevented data-v filter added in #23095 from impacting styles

### LINKED ISSUE #23385: Fix for #23051 broke
### Environment

- Operating System: Linux
- Node Version:     v16.20.0
- Nuxt Version:     3.7.1
- CLI Version:      3.7.3
- Nitro Version:    2.6.2
- Package Manager:  npm@9.4.2
- Builder:          -
- User Config:      experimental
- Runtime Modules:  -
- Build Modules:    -

### Reproduction

You can use my old preproduction from #23051 https://stackblitz.com/edit/github-ydxe3t?file=pages%2Findex.vue

See that the scoped style (margin) applies to the component. However, if you update Nuxt to 3.7.3, the style no longer applies, since the data-v attribute was removed. 

### Describe the bug

The issue #23051 was fixed in #23095.

However, this prevented scoped style from being applied to the component. 

### Additional context

Sorry for the delayed feedback. As I noted in https://github.com/nuxt/nuxt/pull/23095#issuecomment-1712583276 the removal of data-v attribute entirely lead to styles not being reproduced. I already debugged the issue and I'm going to create a PR with a simple suggested fix.

This PR will revert the expected behavior of scoped styles as in 3.7.1 while keeping @huang-julien 's fix for 404 on soft navigation due to different component hash.

### Logs

_No response_

### PR DESCRIPTION
**Pull request by arsengoian**
### 🔗 Linked issue

Fix #23385

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Fixes #23385
The data-v attribute is only removed for the hash function.
Please validate, this is my first PR 😉

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/nuxt/src/app/components/nuxt-island.ts [hunk 1]
```diff
@@ -50,7 +50,8 @@ export default defineComponent({
     const error = ref<unknown>(null)
     const config = useRuntimeConfig()
     const nuxtApp = useNuxtApp()
-    const hashId = computed(() => hash([props.name, props.props, props.context, props.source]))
+    const propsForHash = props.props ? Object.fromEntries(Object.entries(props.props).filter(([key]) => !key.startsWith('data-v-'))) : {}
```
Comment (huang-julien) [inline packages/nuxt/src/app/components/nuxt-island.ts:53]: This won't be reactive, you need to move it into the computed or to make it computed

**Review by huang-julien — CHANGES_REQUESTED** [review]

File: packages/nuxt/src/app/components/nuxt-island.ts (hunk 1 above)
Comment (arsengoian) [reply packages/nuxt/src/app/components/nuxt-island.ts]: Good point

**Comment by arsengoian** [comment]
Will this work?

File: packages/nuxt/src/app/components/nuxt-island.ts [hunk 2]
```diff
@@ -50,7 +50,8 @@ export default defineComponent({
     const error = ref<unknown>(null)
     const config = useRuntimeConfig()
     const nuxtApp = useNuxtApp()
-    const hashId = computed(() => hash([props.name, props.props, props.context, props.source]))
+    const propsForHash = computed(() => props.props ? Object.fromEntries(Object.entries(props.props).filter(([key]) => !key.startsWith('data-v-'))) : {})
+    const hashId = computed(() => hash([props.name, propsForHash.value, props.context, props.source]))
```
Comment (danielroe) [inline packages/nuxt/src/app/components/nuxt-island.ts:54]:
```suggestion
    const filteredProps = computed(() => props.props ? Object.fromEntries(Object.entries(props.props).filter(([key]) => !key.startsWith('data-v-'))) : {})
    const hashId = computed(() => hash([props.name, filteredProps.value, props.context, props.source]))
```

File: packages/nuxt/src/components/runtime/server-component.ts [hunk 3]
```diff
@@ -11,8 +11,7 @@ export const createServerComponent = (name: string) => {
         return h(NuxtIsland, {
           name,
           lazy: props.lazy,
-          // #23051 - remove data-v attributes
-          props: Object.fromEntries(Object.entries(attrs).filter(([key]) => !key.startsWith('data-v-')))
+          props: attrs
```
Comment (danielroe) [inline packages/nuxt/src/components/runtime/server-component.ts:14]:
```suggestion
          props: filteredProps.value
```

**Review by huang-julien — COMMENTED** [review]
I also think that you'll need to update the test in tests/nuxt/nuxt-island.test.ts, it will probably break

**Comment by arsengoian** [comment]
Yeah, just saw that too. Will do

**Review by huang-julien — APPROVED** [review]
lgtm :+1: will probably try to implement nuxt-island runtime test later

**Pull request merged**